### PR TITLE
feat: parse inline config with solang

### PR DESCRIPTION
**Pull request by DaniPopes**
Fixes https://github.com/foundry-rs/foundry/issues/7310

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
great!

pending @klkvr

**Comment by DaniPopes** [comment]
with https://github.com/foundry-rs/foundry/pull/7431/commits/a3b20da361425b08b53381bec688ce688baf0748 this has virtually no perf impact even on huge repos

File: crates/config/src/inline/natspec.rs [hunk 1]
```diff
@@ -28,14 +29,27 @@ impl NatSpec {
...
+        let solang = SolangParser::new();
         for (id, artifact) in output.artifact_ids() {
-            let Some(ast) = &artifact.ast else { continue };
-            let path = id.source.as_path();
-            let path = path.strip_prefix(root).unwrap_or(path);
-            // id.identifier
+            let abs_path = id.source.as_path();
+            let path = abs_path.strip_prefix(root).unwrap_or(abs_path);
+            // `id.identifier` but with the stripped path.
             let contract = format!("{}:{}", path.display(), id.name);
```
Comment (klkvr) [inline crates/config/src/inline/natspec.rs:38]: id.name can be in the form of "{Counter}.0.8.23", we should probably fix this in compilers, I am doing `id.name.split('.').next().unwrap()` usually

**Review by klkvr — APPROVED** [review]
nit, lgtm overall

those are not working correctly with inheritance (and never really worked), right?

**Comment by DaniPopes** [comment]
> those are not working correctly with inheritance (and never really worked), right?

Yes

**Comment by mds1** [comment]
> those are not working correctly with inheritance (and never really worked), right?

Does inline config not work at all with inheritance, or it just doesn't properly override it? For either answer we should just document any limitations in the book https://book.getfoundry.sh/reference/config/inline-test-config#in-line-invariant-configs

**Pull request merged**